### ISSUE TITLE
[Gen3 WS-HPG v1.0 Seed42] Validation-only Direction-1 replacement gate

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#70]
## Objective

Authorize exactly one **seed42 validation-only WS-HPG v1.0 scientific run** after accepted implementation PR #66 and accepted training-preparation PR #69.

This run answers one question only:

> Can the preregistered weak-support hierarchical graph, whose representation originates directly from raw local pixel neighborhoods, outperform the accepted LAP Direction-1 validation endpoint while reducing overfit and avoiding strong dependence on the MediaPipe support prior?

This is a single-seed replacement **gate**, not final replacement proof. No test access is authorized.

## Exact execution source

Run only from exact merged main commit:

`ab7c7a49e923764b6192d9e774352fddf8f1df8b`

Accepted architecture source locks:

- `model.py` SHA-256 `177a782cd8d5c2178303c44d120dcdd22b0a2108a0b720c5091a19f3d7cbffe3`
- `support.py` SHA-256 `b6ed2ddd20a4e82824208929709ff2d6bcb1c5557144d778ed0157fd4768aeee`
- parameters `707,213`
- trainable variable objects `118`
- total Keras variables `138`

Training adapter source locks:

- `__init__.py` `b211b99534a7c3e9ff57ea5ac0b2a7f059307cb56d49f9f1e1214189e6402f43`
- `augmentation.py` `cd89a727a2fb0037dad6a51da87227101f4d51f9e6a8945f5ccca223d66967cd`
- `data.py` `c9a0310a2dcda5eab779366b7c2a9357961c273ea2279525ccfb6453f79823b7`
- `train_validation_only.py` `5b84f29703bc53fbcf5941ec3abbd3af434c6b289156a56845541b35b9fadbe9`

Any source or identity mismatch => STOP before FER training; attempt not consumed.

## Direction-1 scientific identity

The architecture must be interpreted as:

`raw local pixel neighborhoods -> local relational reasoning -> deterministic spatial composition -> coarsened structural relations -> expression`

It is NOT a learned facial-region method.

Locked boundaries:

- raw non-overlapping `3x3` grayscale micro-patches;
- no CNN/pretrained visual backbone before graph reasoning;
- no ROI/RoIAlign;
- no learned spatial-attention region tokens;
- no anatomical/eye/mouth/brow/nose/AU node identity;
- no semantic-region discovery;
- deterministic `2x2` hierarchy is spatial composition;
- mid/coarse dynamic-kNN nodes are `coarsened structural units`, never semantic facial regions;
- MediaPipe only attenuates fine-stage communication and is absent after Pool 1.

## Data

Locked FER2013 split sizes:

- train `28,709`
- validation `3,589`
- test `3,589` and absolutely forbidden.

Registered CSV paths:

- train `/kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv`
- validation `/kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv`

Registered MediaPipe detector-output cache root:

`/kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue`

WS-HPG may read only these NPZ fields:

- `sample_index`
- `label`
- `image_48`
- `detected`
- `landmark_xy_48`

Forbidden semantic arrays must never be read:

- `face_mask`
- `part_soft_masks`
- `micro_anchor_maps`
- `distance_maps`
- `valid_part_mask`
- `valid_anchor_mask`
- `quality_score`

No test file/path/hash/listing/discovery is permitted.

## Support semantics

`landmark_xy_48` is used only as cached output of the accepted detector in FER pixel coordinates `[0,47]`.

Coordinates are immediately reduced to global face extent and the accepted soft support ellipse. Individual landmark identity does not enter the model.

Missing/invalid/degenerate detector output => all-ones support = NO PRIOR.

No support dropout.

## Registered augmentation

Exact accepted ordering:

`original sample identity -> seed42 shuffle with reshuffle_each_iteration=True -> post-shuffle enumerate -> stateless augmentation from enumerated position`

Original identity remains only for cache/image/label/support alignment.

Magnitudes remain frozen:

- horizontal flip `p=0.5`
- rotation uniform `[-10,+10]` degrees
- x/y translation uniform `[-4,+4]` pixels
- contrast `[0.85,1.15]`
- normalized brightness `[-0.10,+0.10]`
- random erasing `p=0.25`
- erase area `[0.02,0.10]`
- erase aspect `[0.5,2.0]`

Image/support share exactly the same geometric affine transform. Photometric operations and erasing are image-only. MediaPipe is not rerun after augmentation.

## Frozen training protocol

Exactly:

- seed `42`
- AdamW
- learning rate `3e-4`
- weight decay `5e-4`
- global clip norm `1.0`
- batch size `64`
- max epochs `100`
- linear warmup `5` epochs
- cosine decay to `1e-6`
- training label smoothing `0.05`
- validation every epoch
- checkpoint = earliest strict maximum `val_accuracy`
- early stop monitor = Keras fit `val_loss`
- patience `15`
- min_delta `0.0`
- mixed precision `false`
- XLA `false`
- MirroredStrategy `false`
- support dropout `false`

Important exact-loss semantics:

- `model.fit` is compiled with the registered LS0.05 training loss; therefore its per-epoch validation `val_loss` used by early stopping is the same smoothed categorical CE under Keras validation evaluation;
- checkpoint selection is by validation accuracy, not validation loss;
- after training, the selected checkpoint is reloaded with `compile=False` and the registered final clean-train / validation losses are **hard CE** through the explicit evaluator.

No post-hoc change to these semantics is allowed.

## Registered execution runtime

Use a Kaggle T4 runtime with exact software:

- Linux
- Python `3.12.12`
- TensorFlow `2.18.1`
- Keras `3.15.0`
- CUDA `12.5.1`
- cuDNN major `9`

Scientific process must see exactly **one Tesla T4**. If the Kaggle host exposes T4x2, launch the scientific process with `CUDA_VISIBLE_DEVICES=0` set before Python/TensorFlow import. MirroredStrategy remains false.

Default floating-point policy must be `float32`. Do not enable mixed precision, XLA, or op determinism as a runtime amendment.

Before FER I/O, run a synthetic batch-64 forward/backward GPU preflight on the exact source/runtime. Require finite loss/gradients and no OOM. This preflight is technical only and does not consume the scientific attempt.

If the exact runtime or batch-64 synthetic preflight fails, STOP before training; do not silently change batch/runtime/config.

## Exact scientific command

From exact detached checkout, after source/runtime/synthetic preflight:

```bash
CUDA_VISIBLE_DEVICES=0 python -m research.candidates.tf_ws_hpg_v1_training.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --prior-root /kaggle/input/datasets/irthn1311/d16-mediapipe-pixel-priors-best-retry-rescue/outputs/d16_mediapipe_pixel_priors_best_retry_rescue \
  --output-root /kaggle/working/ws_hpg_v1_seed42
```

Once `candidate.fit(...)` begins the first optimizer-bearing training epoch, the seed42 attempt is consumed.

## Selected-checkpoint endpoint

Selected checkpoint = earliest strict global maximum validation accuracy.

After normal/registered early-stop completion, reload the selected `.keras` with `compile=False` and use the same loaded object/weights for exactly:

1. clean full train, normal support;
2. validation, normal support;
3. validation, all-ones support.

Final reported loss for these three evaluations is hard CE.

Report:

- selected epoch;
- epochs completed;
- clean-train accuracy / macro-F1 / hard-CE loss;
- validation normal-support accuracy / macro-F1 / hard-CE loss;
- validation all-ones accuracy / macro-F1 / hard-CE loss;
- accuracy gap;
- macro-F1 gap;
- signed support dependency accuracy pp;
- signed support dependency macro-F1 pp;
- support coverage mean/std;
- detector failure count/rate.

Support diagnostics cannot affect training or checkpoint selection.

## Locked comparators

Primary Direction-1 LAP comparator:

- val accuracy `0.6319308999721371`
- val macro-F1 `0.5938407974340496`
- clean-train macro-F1 `0.8207655611897143`
- macro gap `22.69247637556647 pp`

Secondary CF-HPG v1.2 comparator:

- val accuracy `0.5806631373641683`
- val macro-F1 `0.5238975323290902`
- clean-train accuracy `0.614894284022432`
- clean-train macro-F1 `0.5614369765915708`

Issue #68 LAP-LS0.05 outcome, if it finishes before this run, is **descriptive context only** and must NOT change WS-HPG thresholds or architecture/training protocol.

## Exact seed42 decision precedence

Automatic classifier uses the implementation as merged. `capacity_limited` remains its default `False`; it MUST NOT be flipped post-hoc after looking at curves.

1. `WS_HPG_V1_STRETCH_REPLACEMENT_CANDIDATE`
2. `WS_HPG_V1_REPLACEMENT_CANDIDATE`
3. `WS_HPG_V1_COMPETITIVE_GENERALIZATION_SIGNAL_NOT_REPLACE`
4. `WS_HPG_V1_BEATS_CF_ONLY_NOT_REPLACE`
5. `WS_HPG_V1_UNDERFIT_OR_REGRESSION`
6. `WS_HPG_V1_INCONCLUSIVE`

### STRETCH_REPLACEMENT_CANDIDATE
All:

- validation accuracy `>= 0.6500`
- validation macro-F1 `>= 0.6200`
- macro gap `<= 10.0 pp`
- signed support-dependency accuracy `<= 3.0 pp`
- signed support-dependency macro-F1 `<= 3.0 pp`

### REPLACEMENT_CANDIDATE
Unless stretch applies, all:

- validation accuracy `>= 0.6369308999721371`
- validation macro-F1 `>= 0.6038407974340496`
- macro gap `<= 15.0 pp`
- signed support-dependency accuracy `<= 5.0 pp`
- signed support-dependency macro-F1 `<= 5.0 pp`

### COMPETITIVE_GENERALIZATION_SIGNAL_NOT_REPLACE
Unless higher priority applies, all:

- validation accuracy `>= 0.6269308999721371`
- validation macro-F1 `>= 0.5888407974340496`
- macro-gap improvement vs LAP `>= 7.5 pp`

### BEATS_CF_ONLY_NOT_REPLACE
Unless higher priority applies:

- validation accuracy `> 0.5806631373641683`
- validation macro-F1 `> 0.5238975323290902`

but LAP replacement criteria are not met.

### UNDERFIT_OR_REGRESSION
Unless higher priority applies:

- validation accuracy `<= 0.5806631373641683`
  OR
- validation macro-F1 `<= 0.5238975323290902`.

Else `INCONCLUSIVE`.

Only either replacement-candidate status can authorize a later preregistered multi-seed confirmation. A single seed42 outcome never automatically replaces LAP in the paper.

## Validity gates

A scientific decision is valid only if:

- exact checkout/source hashes PASS;
- exact model identity PASS;
- registered runtime PASS;
- synthetic batch64 GPU preflight PASS before FER I/O;
- subprocess completes normally or registered early stopping;
- selected checkpoint exists and reloads `compile=False`;
- selected epoch agrees with earliest strict global max val accuracy;
- selected weights remain byte-identical before/after final diagnostics;
- final evaluation inventory is exactly clean-train normal / validation normal / validation ones;
- all final metrics finite;
- no source mutation;
- no test access or test artifact;
- no hyperparameter or architecture change.

If any validity gate fails after training starts:

`scientific_result_valid=false`

`scientific_interpretation=null`

No automatic retry is authorized.

## Evidence

Capture exact checkout, source hashes, runtime, GPU identity, environment policy, preflight result, exact command, stdout/stderr, training completion reason, history, selected checkpoint identity, final result JSON, and test-access scan.

Package final evidence once; report ZIP SHA-256/bytes/member count/integrity and embedded `.keras` SHA-256/member count/integrity. Do not repack after final hash.

Final valid status:

`WS_HPG_V1_SEED42_COMPLETE_PENDING_INDEPENDENT_REVIEW`

Invalid status:

`WS_HPG_V1_SEED42_INVALID_PENDING_INDEPENDENT_REVIEW`

## Interpretation boundary

A positive result supports only the registered Direction-1 replacement hypothesis under this architecture/training protocol. It does not prove MediaPipe is generally unnecessary, does not establish statistical superiority from one seed, and does not authorize test access.
